File: leetcode/209.长度最小的子数组/209.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0;
        int substring = 0;
        int result = INT32_MAX;
        int i = 0;
        for (int j = 0; j < nums.size(); j++) {
            sum += nums[j];
            while (sum >= target) {
                substring = j - i + 1;
                result = result < substring ? result : substring;
                sum -= nums[i++];
            }
        }
        return result == INT32_MAX ? 0 : result;
    }
};

File: leetcode/209.长度最小的子数组/209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sumvalue, i, substring, result = 0, 0, 0, float('inf')
        for j in range(len(nums)):
            sumvalue = sumvalue + nums[j]
            while sumvalue >= target:
                substring = j - i + 1
                result = result if result < substring else substring
                sumvalue = sumvalue - nums[i]
                i = i + 1
        return 0 if result == float('inf') else result